contracts/cw20-merkle-airdrop/src/contract.rs

```diff
@@ -0,0 +1,487 @@
+#[cfg(not(feature = "library"))]
+use cosmwasm_std::entry_point;
+use cosmwasm_std::{
+    attr, to_binary, Addr, Binary, Deps, DepsMut, Env, MessageInfo, Response, StdResult, SubMsg,
+    Uint128, WasmMsg,
+};
+
+use crate::state::{Config, CLAIM, CONFIG, MERKLE_ROOT, STAGE};
+
+use crate::error::ContractError;
+use crate::msg::{
+    ConfigResponse, ExecuteMsg, InstantiateMsg, IsClaimedResponse, LatestStageResponse,
+    MerkleRootResponse, MigrateMsg, QueryMsg,
+};
+use cw20::Cw20ExecuteMsg;
+use cw_storage_plus::U8Key;
+use sha3::Digest;
+use std::convert::TryInto;
+
+// Version info, for migration info
+const _CONTRACT_NAME: &str = "crates.io:cw20-merkle-airdrop";
+const _CONTRACT_VERSION: &str = env!("CARGO_PKG_VERSION");
+
+#[cfg_attr(not(feature = "library"), entry_point)]
+pub fn instantiate(
+    deps: DepsMut,
+    _env: Env,
+    _info: MessageInfo,
+    msg: InstantiateMsg,
+) -> StdResult<Response> {
+    let config = Config {
+        owner: deps.api.addr_validate(&msg.owner)?,
```

Comment:
You could also:

1) Remove owner and use `info.sender` as the owner, as that often makes sense and is simpler
2) Make owner an Option, use it if provided, or default to `info.sender`.

Both seem slightly nicer for usability. But this is fine for correctness